TheFoodieNetwork/app/src/main/java/com/fbu/thefoodienetwork/activities/SignUpActivity.java

```diff
@@ -0,0 +1,136 @@
+package com.fbu.thefoodienetwork.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Log;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.fbu.thefoodienetwork.R;
+import com.fbu.thefoodienetwork.databinding.ActivitySignUpBinding;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class SignUpActivity extends AppCompatActivity {
+    private static final String TAG = "SignUpActivity";
+    private static final int PASS_UNMATCH = -1;
+    private static final int PASS_INVALID_LENGTH = 1;
+    private static final int PASS_VALID = 0;
+
+    private static EditText usernameEditText;
+    private static EditText emailEditText;
+
+    private static boolean isValidEmail(CharSequence target) {
+        if (target == null) {
+            return true;
+        }
+        return (!TextUtils.isEmpty(target) && Patterns.EMAIL_ADDRESS.matcher(target).matches());
+    }
+
+    private static int isValidPassword(String password, String confirmPass) {
+        if (password.length() < 6) {
+            return PASS_INVALID_LENGTH;
+        }
+        if (!password.equals(confirmPass)) {
+            return PASS_UNMATCH;
+        }
+        return PASS_VALID;
```

Comment:
Something that we could do to make this more thorough is to use regex like you are in the email validation to make sure we're meeting specific requirements.

Here's what that might look like: `((?=.*[a-z])(?=.*d)(?=.*[@#$%])(?=.*[A-Z]).{6,16})`
Here's the breakdown of what this means: https://howtodoinjava.com/regex/how-to-build-regex-based-password-validator-in-java/